moveit_simple/include/moveit_simple/online_robot.h

```diff
@@ -91,6 +138,9 @@ class OnlineRobot : public Robot
   moveit_visual_tools::MoveItVisualToolsPtr online_visual_tools_;
   actionlib::SimpleActionClient<control_msgs::FollowJointTrajectoryAction> action_;
   ros::Subscriber j_state_sub_;
+
+  control_msgs::FollowJointTrajectoryGoal execution_goal_;
+  bool execute_collision_check_ = false;
```

Comment:
This seems like an odd flag to keep around.  Does it need to be a class variable (is it used by multiple class members, does it's value need to be persistent beyond function calls).

This is likely to cause bugs in the future.

Reply:
> Does it need to be a class variable (is it used by multiple class members, does it's value need to be persistent beyond function calls).

Yes, it is used by multiple class members. The value is persistent between goals, so if a new goal is set the value gets changed.